vexination/walker/src/server.rs

```diff
@@ -7,32 +7,48 @@ use csaf_walker::{
     validation::{ValidatedAdvisory, ValidationError, ValidationOptions, ValidationVisitor},
     walker::Walker,
 };
+use reqwest::StatusCode;
 use serde::Deserialize;
 use tokio::sync::{Mutex, RwLock};
+use trustification_auth::client::TokenInjector;
+use trustification_auth::client::TokenProvider;
 
 pub async fn run(
     workers: usize,
     source: url::Url,
     sink: url::Url,
+    provider: Arc<dyn TokenProvider>,
     options: ValidationOptions,
 ) -> Result<(), anyhow::Error> {
     let fetcher = Fetcher::new(Default::default()).await?;
 
     let validation = ValidationVisitor::new(|advisory: Result<ValidatedAdvisory, ValidationError>| {
         let sink = sink.clone();
+        let provider = provider.clone();
         async move {
             match advisory {
                 Ok(ValidatedAdvisory { retrieved }) => {
                     let data = retrieved.data;
                     match serde_json::from_slice::<csaf::Csaf>(&data) {
-                        Ok(doc) => match reqwest::Client::new().post(sink).json(&doc).send().await {
-                            Ok(_) => {
+                        Ok(doc) => match reqwest::Client::new()
```

Comment:
I now it has nothing to do with this PR, but I think there should be a single client which gets reused. 

Citing the docs:

> **NOTE:** If you plan to perform multiple requests, it is best to create a [Client](https://docs.rs/reqwest/0.11.20/reqwest/struct.Client.html) and reuse it, taking advantage of keep-alive connection pooling.

https://docs.rs/reqwest/0.11.20/reqwest/#making-a-get-request